pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -81,7 +81,7 @@
 public class MultiStageBrokerRequestHandler extends BaseBrokerRequestHandler {
   private static final Logger LOGGER = LoggerFactory.getLogger(MultiStageBrokerRequestHandler.class);
 
-  private final QueryEnvironment _queryEnvironment;
+  private final WorkerManager _workerManager;
```

Comment:
Any specific reason why we cannot share the `QueryEnvironment`? Is it stateful?

Reply:
Do we need to keep it? It seems it is only be used in _handleRequest_

Reply:
Right now when a new database is created its not getting synced in the catalog hence we need to create the `QueryEnvironment` for each query.

Reply:
Based on the updated code now the `QueryEnvironment` will only contain tables for the database context associated with each query hence it needs to be created for each query.